Products/ZenUtils/Auth0/Auth0Callback.py

```diff
@@ -36,40 +35,53 @@ def __call__(self):
         args = getQueryArgsFromRequest(self.request)
         state_arg = args.get('state')
         code = args.get('code')
+        error = args.get('error', None)
+
+        if error:
+            log.debug('Auth0 error response: {}'.format(error))
 
         domain = conf['tenant'].replace('https://', '').replace('/', '')
 
-        data = {
-            "grant_type": "authorization_code",
-            "client_id": conf['clientid'],
-            "client_secret": conf['client-secret'],
-            "code": code,
-            "audience": "%s/userinfo" % domain,
-            "scope": "openid profile",
-            "redirect_uri": "%s/zport/Auth0Callback" % zenoss_uri
-        }
+        if error:
+            # We need to make a new request to authorize without the
+            # prompt parameter so that auth0 presents a login screen.
+            uri = "%sauthorize?" % conf['tenant'] + \
+                  "response_type=code&" + \
+                  "client_id=%s&" % conf['clientid'] + \
+                  "state=%s&" % state_arg + \
+                  "scope=openid offline_access&" + \
+                  "redirect_uri=%s/zport/Auth0Callback" % zenoss_uri
+            return self.request.response.redirect(uri)
+        else:
```

Comment:
This `else` is not necessary.  The last line of the `if` is a return, and the rest of the code is all in the else clause.  It would make the code simpler and easier to review if we took it out.